activesupport/lib/active_support/file_evented_update_checker.rb

```diff
@@ -0,0 +1,60 @@
+require 'listen'
+
+module ActiveSupport
+  class FileEventedUpdateChecker
+    attr_reader :listener
+    def initialize(files, directories={}, &block)
+      @files = files.map { |f| File.expand_path(f)}.to_set
+      @dirs = Hash.new
+      directories.each do |key,value|
+        @dirs[File.expand_path(key)] = Array(value) if !Array(value).empty?
+      end
+      @block = block
+      @modified = false
+      # Listeners for files.
+      begin
+        @files.each do |file|
+          @listener = Listen.to(File.expand_path("#{file}/.."),&method(:changed))
+          @listener.only(Regexp.new("#{Regexp.quote(File.basename(file))}$"))
+          @listener.start if @listener
+        end
+      rescue
```

Comment:
There may be files which does not exist and if in case the file is already watched then Listen explicitly does not handle that, that's why I have rescued the code.